chef/automate
# iam_projects: drop projects column

### :nut_and_bolt: Description: What code changed, and why?

For some reason, we had wanted to assign a projects one or more projects. Doing that, we've introduced a `projects TEXT[]` database column, with no constraints whatsoever. Disliking that, we've discussed this again, and decided to drop the column. This is what this PR does.

### :chains: Related Resources

#832 

### :+1: Definition of Done

Our "projects" IAM entities no longer have "projects" themselves. There's no database column, and no API response, etc.

### :athletic_shoe: How to Build and Test the Change

- `start_all_services`, `rebuild components/authz-service`, `rebuild components/automate-gateway`
- check direct grpc calls:
```
[13][default:/src:0]# chef-automate dev grpcurl authz-service --  -d "$(jq -n '{id: "foo", name: "Foo" }')" chef.automate.domain.authz.v2.Projects.CreateProject
{
  "project": {
    "name": "Foo",
    "id": "foo",
    "type": "CUSTOM"
  }
}
[14][default:/src:0]# chef-automate dev grpcurl authz-service --  chef.automate.domain.authz.v2.Projects.ListProjects
{
  "projects": [
    {
      "name": "Foo",
      "id": "foo",
      "type": "CUSTOM"
    }
  ]
}
[15][default:/src:0]#
```
- check http requests:
```
[19][default:/src:0]# export TOKEN=$(chef-automate iam token create --admin iam-admin-token)
[20][default:/src:0]# curl -kH "api-token: $TOKEN" https://localhost/apis/iam/v2beta/projects/foo
{"project":{"name":"Foo","id":"foo","type":"CUSTOM"}}
[21][default:/src:0]# curl -kH "api-token: $TOKEN" https://localhost/apis/iam/v2beta/projects
{"projects":[{"name":"Foo","id":"foo","type":"CUSTOM"}]}
[22][default:/src:0]#
```

### :white_check_mark: Checklist
